pkg/agent/run.go

```diff
@@ -98,6 +99,9 @@ func run(ctx context.Context, cfg cmds.Agent, proxy proxy.Proxy) error {
 	if err != nil {
 		return err
 	}
+
+	app.WaitForAPIServer(coreClient, 30*time.Second)
```

Comment:
Is this something we can use in the other places of the app that wait for the api-server to become available?

Reply:
This is the same thing that the server does, except it does it in a goroutine that then sends on a wait channel to unblock the rest of the server startup:

https://github.com/k3s-io/k3s/blob/master/pkg/daemons/control/server.go#L1043

On the agent side, the rest of our code and the kubelet itself just retries as necessary; Flannel seems to be the only thing that crashes if the apiserver isn't ready when it starts up.